File: src/model/class-contest/ClassContestResponse.ts
import { ClassContestStatus } from "@/types/enum/ClassContestStatus";

export interface ClassContestResponse {
  classContestId: string;
  classInfo: {
    classId: string;
    classCode: string;
    className: string;
    instructorName: string;
  };
  contestInfo: {
    contestLessonId: string;
    title: string;
    description: string;
    defaultTotalPoints: number;
    codingExerciseCount: number;
    quizQuestionCount: number;
  };
  scheduledStartTime: string;
  scheduledEndTime: string;
  durationInMinutes: number;
  status: ClassContestStatus;
  isActive: boolean;
  weight: number;
  effectiveConfig: {
    maxAttempts: number;
    showLeaderboard: boolean;
    instructions: string;
    passingScore: number;
    totalPoints: number;
  };
  createdAt: string;
  updatedAt: string;
}

File: src/main/java/com/truongsonkmhd/unetistudy/dto/contest_lesson/ClassContestResponse.java
package com.truongsonkmhd.unetistudy.dto.contest_lesson;


import com.truongsonkmhd.unetistudy.common.ClassContestStatus;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.Instant;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ClassContestResponse {

    UUID classContestId;

    // Thông tin lớp học
    ClassInfo classInfo;

    // Thông tin contest template
    ContestInfo contestInfo;

    // Thời gian thi
    Instant scheduledStartTime;
    Instant scheduledEndTime;
    Long durationInMinutes;

    // Cấu hình
    ClassContestStatus status;
    Boolean isActive;
    Double weight;

    // Cấu hình hiệu quả
    EffectiveConfig effectiveConfig;

    Instant createdAt;
    Instant updatedAt;

}